httpcore5/src/main/java/org/apache/hc/core5/http/impl/io/DefaultResponseOutOfOrderStrategy.java

```diff
@@ -0,0 +1,54 @@
+/*
+ * ====================================================================
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+
+package org.apache.hc.core5.http.impl.io;
+
+import org.apache.hc.core5.http.ClassicHttpRequest;
+import org.apache.hc.core5.http.io.ResponseOutOfOrderStrategy;
+
+/**
+ * Default {@link ResponseOutOfOrderStrategy} implementation which checks for premature responses every 8kb based
+ * on testing using values between 4 KiB and 128 KiB. This is optimized for correctness, and results in a maximum
+ * upload speed of 8 MiB/s.
+ *
+ * @see <a href="https://www.mail-archive.com/httpclient-users@hc.apache.org/msg09911.html">msg09911</a>
+ * @see <a href="https://issues.apache.org/jira/browse/HTTPCORE-639">HTTPCORE-639</a>
+ * @since 5.1
+ */
+public enum DefaultResponseOutOfOrderStrategy implements ResponseOutOfOrderStrategy {
```

Comment:
While this seems to be fine Java-wise, we have never applied this enum pattern to be immutable elsewhere.

Reply:
Happy to migrate this to a final class with a private constructor if you prefer. I tend to prefer enum singletons because it's not necessary to define a private constructor and they cannot be reflectively instantiated, but I agree that it's best to be consistent across the codebase.

Reply:
Please have a look at existing default strategies and apply the same pattern, even if it is suboptimal, but at least consistent as you have said.